toolchain/check/import_ref.cpp

```diff
@@ -1321,6 +1330,20 @@ class ImportRefResolver {
                                         .elements_id = elements_id})};
   }
 
+  auto TryResolveTypedInst(SemIR::IntLiteral inst) -> ResolveResult {
+    auto initial_work = work_stack_.size();
+    auto type_id = GetLocalConstantId(inst.type_id);
+    if (HasNewWork(initial_work)) {
+      return ResolveResult::Retry();
+    }
+    return {.const_id = TryEvalInst(
+                context_, SemIR::InstId::Invalid,
+                SemIR::IntLiteral{
+                    .type_id = context_.GetTypeIdForTypeConstant(type_id),
+                    .int_id = context_.ints().Add(
+                        import_ir_.ints().Get(inst.int_id))})};
```

Comment:
These calls, as a set, sort of feel like they'd wrap better (reading a little more easily) if you used some locals for intermediate work. You could even do something like:

```suggestion
    SemIR::IntLiteral inst = {
        .type_id = context_.GetTypeIdForTypeConstant(type_id),
        .int_id = context_.ints().Add(import_ir_.ints().Get(inst.int_id))};
    return {.const_id = TryEvalInst(context_, SemIR::InstId::Invalid, inst)};
```